src/core/modules/impl/ArweaveGatewayInteractionsLoader.ts

```diff
@@ -116,15 +117,17 @@ export class ArweaveGatewayInteractionsLoader implements InteractionsLoader {
     });
 
     const isLocalOrTestnetEnv = this.environment === 'local' || this.environment === 'testnet';
+    const vrfPlugin = this._warp.maybeLoadPlugin<void, VrfPluginFunctions>('vrf');
+
     return sortedInteractions.map((i) => {
       const interaction = i.node;
       if (isLocalOrTestnetEnv) {
-        if (
-          interaction.tags.some((t) => {
-            return t.name == WARP_TAGS.REQUEST_VRF && t.value === 'true';
-          })
-        ) {
-          interaction.vrf = generateMockVrf(interaction.sortKey, this.arweave);
+        if (hasVrfTag(interaction)) {
+          if (vrfPlugin) {
+            interaction.vrf = vrfPlugin.process().generateMockVrf(interaction.sortKey);
+          } else {
+            this.logger.warn('Cannot generate mock vrf for interaction - no "warp-contracts-plugin-vrf" attached!');
+          }
```

Comment:
hmm, seems tripled. can't we generate a custom function for the vrf generation?
edit: it's even quadrupled